File: ProgramingApp/app/src/main/java/com/example/programingapp/MainActivity.java
package com.example.programingapp;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import com.example.programingapp.notification.RandomQuoteAlarm;
import com.example.programingapp.quotes_details.QuoteDetailFragment;
import com.example.programingapp.quotes_list.QuotesListFragment;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RandomQuoteAlarm randomQuoteAlarm = new RandomQuoteAlarm(this);
        randomQuoteAlarm.setUpAlarm();

        QuotesListFragment fragment = new QuotesListFragment();
        FragmentManager fragmentManager = getSupportFragmentManager();
        fragmentManager
                .beginTransaction()
                .replace(R.id.frameLayout, fragment)
                .commit();


    }
}

File: KotlinFirstExample/app/src/main/java/com/example/kotlinfirstexample/MainActivity.kt
package com.example.kotlinfirstexample

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mainFragment = MainFragment()
        mainFragment.listener = object : NekojListener {
            override fun nekojaMetoda(text: String) {

            }
        }

        mainFragment.listenerFunc = {
            Toast.makeText(this, it, Toast.LENGTH_LONG).show()
        }

        supportFragmentManager.beginTransaction()
            .replace(R.id.main_container, mainFragment)
            .commit()
    }
}

class ListenerImpl : NekojListener {
    override fun nekojaMetoda(text: String) {

    }

}
